luigi/lock.py

```diff
@@ -104,8 +105,14 @@ def acquire_for(pid_dir, num_available=1, kill_signal=None):
 
     # Check if there is a pid file corresponding to this name
     if not os.path.exists(pid_dir):
-        os.mkdir(pid_dir)
-        os.chmod(pid_dir, 0o777)
+        # Try/except for handling race condition
+        try:
+            os.mkdir(pid_dir)
+            os.chmod(pid_dir, 0o777)
+        except OSError as exc:
+            if exc.errno != errno.EEXIST:
+                raise
+            pass
```

Comment:
Why have the `if not os.path.exists(pid_dir)` if this try/except will handle the case when it already exists? While still correct to have both, I believe the Pythonic way is just to try and allow to fail. The try/except should handle both the normal case when the `pid_dir` already exists AND when this race condition occurs.

Reply:
@dlstadther done. It also resolved codecov checks.